lib/ansible/modules/windows/win_psrepository.ps1

```diff
@@ -1,68 +1,229 @@
 #!powershell
 
+# Copyright: (c) 2020, Brian Scholer <@briantist>
 # Copyright: (c) 2018, Wojciech Sciesinski <wojciech[at]sciesinski[dot]net>
 # Copyright: (c) 2017, Daniele Lazzari <lazzari@mailup.com>
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 #Requires -Module Ansible.ModuleUtils.Legacy
+#Requires -Module @{ ModuleName = 'PowerShellGet' ; ModuleVersion = '1.6.0' }
 
 # win_psrepository (Windows PowerShell repositories Additions/Removals/Updates)
 
 $params = Parse-Args -arguments $args -supports_check_mode $true
 $check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
 
 $name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
-$source = Get-AnsibleParam -obj $params -name "source" -type "str"
+$source_location = Get-AnsibleParam -obj $params -name "source_location" -aliases "source","module_source","module_source_location" -type "str"
+$script_source_location = Get-AnsibleParam -obj $params -name "script_source_location" -aliases "script_source" -type "str"
+$publish_location = Get-AnsibleParam -obj $params -name "publish_location" -aliases "module_publish_location" -type "str"
```

Comment:
Same thing here, I would probably just have `publish_location` as that is the default type and only have the `script_*` ones with the prefix.